### PR TITLE
added validation in TeacherDTO & SubjectDTO, fixed error msg in PostM…

### PR DESCRIPTION
**Pull request by osman-jft**
…apping in TeacherController, fixed formatting of code

### CODE REVIEW

File: src/main/resources/application.properties
Comment (jaineetsingh-jft) [inline src/main/resources/application.properties:1]: remove this file

File: src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java [hunk 1]
```diff
@@ -15,26 +17,33 @@ public class TeacherController {
...
     public TeacherController(TeacherService teacherService) {
         this.teacherService = teacherService;
     }
 
     @GetMapping
     private ResponseDTO<List<TeacherDTO>> getTeachers() {
-        //returns list of students from StudentService
+        List<TeacherDTO> teacherDto = teacherService.getAllTeachers();
+        if (teacherDto == null) { // check null condition for list
+            return ResponseDTO.<List<TeacherDTO>>builder()
```
Comment (jaineetsingh-jft) [inline src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java]: Dont't return response DTO directly from here,throw custom exception

File: src/main/java/com/example/sb_assign_16_05_23/service/impl/StudentServiceImpl.java [hunk 2]
```diff
@@ -6,40 +6,40 @@
...
-   ModelMapper mapper;
+    @Autowired
+    ModelMapper mapper;
 
     @Override
     public List<StudentDTO> getAllStudents() {
         List<Student> students = studentRepository.findAll();
-        if(students.isEmpty()){
+        if (students.isEmpty()) {
             return null;
```
Comment (jaineetsingh-jft) [inline src/main/java/com/example/sb_assign_16_05_23/service/impl/StudentServiceImpl.java]: throw custom exception

File: src/main/java/com/example/sb_assign_16_05_23/controller/StudentController.java [hunk 3]
```diff
@@ -22,32 +20,26 @@ public class StudentController {
...
     }
 
     @GetMapping
     private ResponseDTO<List<StudentDTO>> getStudents() {
         List<StudentDTO> studentDto = studentService.getAllStudents();
-        if(studentDto == null){ // check null condition for list
-            return ResponseDTO.<List<StudentDTO>>builder()
-                    .data(null).message(Constants.EMPTY_LIST).status(HttpStatus.NO_CONTENT.value())
-                    .build();
+        if (studentDto == null) { // check null condition for list
```
Comment (jaineetsingh-jft) [inline src/main/java/com/example/sb_assign_16_05_23/controller/StudentController.java]: throw exception and handle in service class

File: src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java [hunk 4]
```diff
@@ -15,26 +17,28 @@ public class TeacherController {
...
         this.teacherService = teacherService;
     }
 
     @GetMapping
     private ResponseDTO<List<TeacherDTO>> getTeachers() {
-        //returns list of students from StudentService
-        return ResponseDTO.<List<TeacherDTO>>builder().data(teacherService.getAllTeachers())
-                .message(Constants.SUCCESS_MSG).status(HttpStatus.OK.value()).build();
+        List<TeacherDTO> teacherDto = teacherService.getAllTeachers();
+        if (teacherDto == null) { // check null condition for list
```
Comment (jaineetsingh-jft) [inline src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java]: throw exception and hanlde in service class

File: src/main/resources/application.properties
Comment (jaineetsingh-jft) [inline src/main/resources/application.properties:1]: remove this file

**Review by jaineetsingh-jft — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties
Comment (osman-jft) [reply src/main/resources/application.properties:1]: done

File: src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java (hunk 4 above)
Comment (osman-jft) [reply src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java]: done

File: src/main/java/com/example/sb_assign_16_05_23/controller/StudentController.java (hunk 3 above)
Comment (osman-jft) [reply src/main/java/com/example/sb_assign_16_05_23/controller/StudentController.java]: done

File: src/main/java/com/example/sb_assign_16_05_23/service/impl/StudentServiceImpl.java (hunk 2 above)
Comment (osman-jft) [reply src/main/java/com/example/sb_assign_16_05_23/service/impl/StudentServiceImpl.java]: done

File: src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java (hunk 1 above)
Comment (osman-jft) [reply src/main/java/com/example/sb_assign_16_05_23/controller/TeacherController.java]: done

File: src/main/resources/application.properties
Comment (osman-jft) [reply src/main/resources/application.properties:1]: done

Comment (jaineetsingh-jft) [inline src/main/resources/application.properties:1]: Delete this file

**Review by jaineetsingh-jft — APPROVED** [review]

**Pull request merged**